Source/FullScreenPokemon.ts

```diff
@@ -4829,6 +4829,55 @@ module FullScreenPokemon {
                 callback);
         }
 
+        /* partyActivate functions
+         */
+
+        /**
+         * Makes sure the Player has a Pokemon in the party that knows the necessary HM move.
+         * 
+         * @param player   
+         * @param thing   
+         *
+         * @todo Eventually add check to make sure Player beat Gym leader needed to use HM
+         */
+        checkPartyActivate(player: IPlayer, thing: IHMSolid): void {
```

Comment:
This should be split into two functions. One that checks if the move is in the party (and gets it if it is), and one that calls its .partyActivate